src/filters/network.rs

```diff
@@ -381,6 +398,7 @@ pub struct NetworkFilter {
     pub opt_domains: Option<Vec<Hash>>,
     pub opt_not_domains: Option<Vec<Hash>>,
     pub redirect: Option<String>,
+    pub redirect_url: Option<String>,
```

Comment:
Sadly we can't just add fields in the middle here; this struct definition is directly used in the `v0` format so we'd have to do the same thing with `#[serde(default)]` and add it as the last field of the struct.

Luckily though, since `redirect` and `redirect_url` are mutually exclusive, we do have another option. We can use the original `redirect` field in both cases, and use an unused bit in `NetworkFilterMask` to signify the redirect semantics. So far bits 15, 26, and 31 are unused - you can go ahead and repurpose bit 15 (there used to be a `$fuzzy` option, but the fuzzy matching logic was never implemented). This approach has the added benefit of not increasing the in-memory size of `NetworkFilter`, which is a valid concern when loading thousands of rules.